### PR TITLE
Add a `cuda_device` option to `predict`.

### LINKED ISSUE #303: `predict` does not have a `cuda_device` option.
If you simply pass `cuda_device` to `load_archive` such as in https://github.com/schmmd/allennlp/commit/889ebd3843dbde0039dcf2ab7f2571691f0e4828 you get the following error.

```
Traceback (most recent call last):
  File "allennlp/run", line 14, in <module>
    main()
  File "/stage/allennlp/commands/__init__.py", line 27, in main
    args.func(args)
  File "/stage/allennlp/commands/predict.py", line 82, in predict
    run(predictor, input_file, output_file, not args.silent)
  File "/stage/allennlp/commands/predict.py", line 59, in run
    result = predictor.predict_json(data)
  File "/stage/allennlp/service/predictors/bidaf.py", line 26, in predict_json
    prediction = self.model.predict_span(question, passage)
  File "/stage/allennlp/models/bidaf.py", line 342, in predict_span
    output_dict = self.forward(**model_input)
  File "/stage/allennlp/models/bidaf.py", line 188, in forward
    embedded_question = self._highway_layer(self._text_field_embedder(question))
  File "/opt/conda/lib/python3.5/site-packages/torch/nn/modules/module.py", line 224, in __call__
    result = self.forward(*input, **kwargs)
  File "/stage/allennlp/modules/text_field_embedders/basic_text_field_embedder.py", line 45, in forward
    token_vectors = embedder(tensor)
  File "/opt/conda/lib/python3.5/site-packages/torch/nn/modules/module.py", line 224, in __call__
    result = self.forward(*input, **kwargs)
  File "/stage/allennlp/modules/token_embedders/token_characters_encoder.py", line 36, in forward
    return self._dropout(self._encoder(self._embedding(token_characters), mask))
  File "/opt/conda/lib/python3.5/site-packages/torch/nn/modules/module.py", line 224, in __call__
    result = self.forward(*input, **kwargs)
  File "/stage/allennlp/modules/time_distributed.py", line 29, in forward
    reshaped_outputs = self._module(*reshaped_inputs)
  File "/opt/conda/lib/python3.5/site-packages/torch/nn/modules/module.py", line 224, in __call__
    result = self.forward(*input, **kwargs)
  File "/stage/allennlp/modules/token_embedders/embedding.py", line 116, in forward
    sparse=self.sparse)
  File "/opt/conda/lib/python3.5/site-packages/torch/nn/functional.py", line 632, in embedding
    scale_grad_by_freq, sparse
  File "/opt/conda/lib/python3.5/site-packages/torch/nn/_functions/thnn/sparse.py", line 53, in forward
    output = torch.index_select(weight, 0, indices.view(-1))
TypeError: torch.index_select received an invalid combination of arguments - got (torch.cuda.FloatTensor, int, torch.LongTensor), but expected (torch.cuda.FloatTensor source, int dim, torch.cuda.LongTensor index)
```

### PR DESCRIPTION
**Pull request by schmmd**
This removes some brittle code that only works on machines with 1 GPU and lets you make predictions with a GPU.

On a `p2.xlarge` with a GPU a BiDAF prediction takes 0.20 seconds.  Without a GPU on the same machine a prediction takes 0.44 seconds.  Not a huge difference, because the code is set up to make one prediction at a time.

Fixes #303.

### CODE REVIEW

**Comment by schmmd** [comment]
Fixes https://github.com/allenai/allennlp/issues/303

**Comment by matt-gardner** [comment]
@schmmd, just FYI, I added your "Fixes" comment to the initial description.  For some reason, github only links the PR and the issue if it shows up in the initial description, not in the comments.

File: kubernetes-dev-environment.yaml [hunk 1]
```diff
@@ -2,10 +2,10 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: allennlp-dev-env # TODO: make this unique by appending your username
+  name: allennlp-michaels-env # TODO: make this unique by appending your username
```
Comment (matt-gardner) [inline kubernetes-dev-environment.yaml]: These changes don't belong here, right?

File: allennlp/models/model.py [hunk 2]
```diff
@@ -111,7 +111,6 @@ def forward_on_instance(self, instance: Instance) -> Dict[str, numpy.ndarray]:
         # Hack to see what cuda device the model is on, so we know where to put these inputs.  For
         # complicated models, or machines with multiple GPUs, this will not work.  I couldn't find
         # a way to actually query what device a tensor / parameter is on.
-        cuda_device = 0 if next(self.parameters()).is_cuda else -1
```
Comment (matt-gardner) [inline allennlp/models/model.py]: If you're removing this line, also remove the comments above it.

File: allennlp/models/model.py [hunk 3]
```diff
@@ -100,7 +100,7 @@ def forward(self, input1, input2, targets=None):
         """
         raise NotImplementedError
 
-    def forward_on_instance(self, instance: Instance) -> Dict[str, numpy.ndarray]:
+    def forward_on_instance(self, instance: Instance, cuda_device: int) -> Dict[str, numpy.ndarray]:
```
Comment (matt-gardner) [inline allennlp/models/model.py]: Is `predict` the only command that uses this method?  Does `evaluate` use it?  Or are there any changes necessary in the server code to handle this?

Comment (schmmd) [reply allennlp/models/model.py]: One day @matt-gardner, I'll learn the fact that Python is an interpreted language.

**Comment by schmmd** [comment]
@matt-gardner @joelgrus take another look? Tests pass, linter passes--sorry for the early train wreck.

**Comment by joelgrus** [comment]
nice

**Pull request merged**